src/Swarm/Game/State.hs

```diff
@@ -807,8 +818,16 @@ buildWorld em (WorldDescription {..}) = (robots, first fromEnum . wf)
 initGameStateForScenario :: String -> Maybe Seed -> Maybe String -> ExceptT Text IO GameState
 initGameStateForScenario sceneName userSeed toRun = do
   g <- initGameState
-  scene <- loadScenario sceneName (g ^. entityMap)
-  liftIO $ scenarioToGameState scene userSeed toRun g
+  (scene, path) <- loadScenario sceneName (g ^. entityMap)
+  gs <- liftIO $ scenarioToGameState scene userSeed toRun g
+  normalPath <- liftIO $ normalizeScenarioPath (gs ^. scenarios) path
+  t <- liftIO getZonedTime
+  liftIO . appendFile "/tmp/debug" $ "TODO: S - Scenario in progress " <> normalPath <> "\n"
+  liftIO . appendFile "/tmp/debug" $ "TODO: S - Originally " <> path <> "\n"
+  return $
+    gs
+      & currentScenarioPath ?~ normalPath
+      & scenarios . scenarioItemByPath normalPath . _SISingle . _2 . scenarioStatus .~ InProgress t 0
```

Comment:
It seems like this code is repeated several times and should probably be factored out somehow.  Relatedly, I am not 100% sure why we need a separate `initGameStateForScenario` function.  Is it so we avoid having to load all the scenarios?  I honestly don't really remember.  I think it only gets used in `classicGame0` below, which in turn is only used in the test and benchmark suites.  It seems like we should be able to refactor and delete a bunch of code somehow.

Reply:
Oh yes, I discovered these places incrementally, so I just copy-pasted to get it working.

I think it's just that the initialization functions do not compose well, some take a string, some maybe string and some scenario. It used to not matter because the functions were short, but now it would be annoying to keep in sync.

I will try to refactor this a bit, it should be a small change.